src/PowerShellEditorServices/Logging/HostLoggerAdapter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PowerShellEditorServices.Logging
+{
+    /// <summary>
+    /// Adapter class to allow logging events sent by the host to be recorded by PSES' logging infrastructure.
+    /// </summary>
+    internal class HostLoggerAdapter : IObserver<(int logLevel, string message)>
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Create a new host logger adapter.
+        /// </summary>
+        /// <param name="loggerFactory">Factory to create logger instances with.</param>
+        public HostLoggerAdapter(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger("HostLogs");
+        }
+
+        public void OnCompleted()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)